plugins/primus_lisp/primus_lisp_primitives.ml

```diff
@@ -92,6 +92,28 @@ module MemoryAllocate(Machine : Primus.Machine.S) = struct
     | _ -> Lisp.failf "allocate requires two arguments" ()
 end
 
+module MemoryFill(Machine : Primus.Machine.S) = struct
+  include Lib(Machine)
+  let negone = Value.one 8
+  let zero = Value.zero 8
+
+  let run = function
+    | [addr; len; value] ->
+       let len = Value.to_word len |> Word.to_int in
+       let value = Value.to_word value |> Word.to_int in
+       if Result.is_error len || Result.is_error value
+       then negone
+       else
+         let len = Or_error.ok_exn len in
+         let value = Or_error.ok_exn value in
+         Eval.fill addr len value >>= fun () ->
```

Comment:
remove all this error handing, as we should use values here. 
Also, this error handling is undocumented and is not done correctly...